### PR TITLE
Feat/invt cpc 889 Add Test And Fixed Search Issues

### PR DESCRIPTION
**Pull request by Matthewtan9**
JIRA: link to jira ticket
[## Context:](https://champlainsaintlambert.atlassian.net/browse/CPC-899)
What is the ticket about and why are we doing this change
-Added integration controller testing for regex and paging logic.
-Fixed a bug where it would only show the letter and not the entirety of the searched name. So now if i can do letter, number or name.
What are the various changes and what other modules do those changes affect?
- Covers 100% of the method that i updated so jacoco shows better test results

## Linked pull requests (Optional)
- pull request link

### CODE REVIEW

File: inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java [hunk 1]
```diff
@@ -251,14 +279,26 @@ public Mono<Void> deleteInventoryByInventoryId(String inventoryId) {
...
-            char firstChar = inventoryDescription.charAt(0);
-            return inventoryRepository
-                    .findByInventoryDescriptionRegex("(?i)^" + firstChar + ".*")
-                    .map(EntityDTOUtil::toInventoryResponseDTO)
-                    .skip(page.getPageNumber() * page.getPageSize())
-                    .take(page.getPageSize())
-                    .switchIfEmpty(Mono.error(new NotFoundException("Inventory not found with Description: " + inventoryDescription)));
+        if (inventoryDescription != null) {
+
+            String escapedInventoryDescription = Pattern.quote(inventoryDescription);
```
Comment (Eli293) [inline inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java:284]: Could you explain this code?

Comment (Matthewtan9) [reply inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java:284]: For sure, there is a if the inventoryDescription -= meaning if it's not null, in simpler it checks whether the inventoryDescription has a value or it's an empty string.
For the String Pattern.Quote It is from a java.util.package, Its job is to make sure that any characters in the string, might have some special meaning or regular characters.

**Review by Eli293 — DISMISSED** [review]
Looks good, nice job.

**Review by SandwichKing978 — DISMISSED** [review]
Looks
Good, great job on making it work

**Review by SandwichKing978 — DISMISSED** [review]
Looks good, tests are well done

**Review by SandwichKing978 — APPROVED** [review]

**Review by Eli293 — APPROVED** [review]

**Review by Eli293 — APPROVED** [review]

**Review by SandwichKing978 — APPROVED** [review]

**Pull request merged**